server/src/main/java/org/elasticsearch/action/admin/cluster/node/hotthreads/NodesHotThreadsRequest.java

```diff
@@ -29,7 +30,7 @@ public NodesHotThreadsRequest(StreamInput in) throws IOException {
         super(in);
         threads = in.readInt();
         ignoreIdleThreads = in.readBoolean();
-        type = in.readString();
```

Comment:
This isn't going to be wire compatible, as readEnum uses ordinals for serialization, not the string name. We will need to version this change, using `in.getVersion().onOrAfter(Version.V_7_16_0)` as a condition for the readEnum call (assuming we will backport to 7.x).

Another thing to consider is that because readEnum uses ordinals, any change in the ordering of the ReportType values could cause serialization changes. It might be beneficial to (separately) look at making readEnum use strings internally, but then use `valueOf` to get the enum value from string.

Reply:
Thanks for this find! I didn't think of the mix cluster case and I thought handling it at the REST layer will be enough. I'll revert the request type to String to be backwards compatible and use the .of helper to convert to the enum.